apps/gitness/src/pages-v2/pull-request/pull-request-conversation.tsx

```diff
@@ -195,31 +278,35 @@ export default function PullRequestConversationPage() {
     [updateTitle, refetchPullReq]
   )
 
-  const onDeleteBranch = () => {
+  const onDeleteBranch = useCallback(() => {
+    const onSuccessDeleteCommon = () => {
+      setShowDeleteBranchButton(false)
+      setShowRestoreBranchButton(true)
+    }
+
     deleteBranch({
       repo_ref: repoRef,
       pullreq_number: prId,
       queryParams: { bypass_rules: true, dry_run_rules: false }
     })
       .then(res => {
         refetchBranch()
+
         if (res?.body?.rule_violations) {
           const { checkIfBypassAllowed } = extractInfoFromRuleViolationArr(res.body?.rule_violations)
-          if (checkIfBypassAllowed) {
-            setShowDeleteBranchButton(true)
-          } else {
-            setShowDeleteBranchButton(false)
-            setShowRestoreBranchButton(true)
-          }
+
+          checkIfBypassAllowed ? setShowDeleteBranchButton(true) : onSuccessDeleteCommon()
```

Comment:
I'm not a fan of using ternaries as a substitute for `if` when invoking functions.
```suggestion
          if (checkIfBypassAllowed) {
            setShowDeleteBranchButton(true)
          } else {
            onSuccessDeleteCommon()
          }
```

Not a blocker, I just don't find it as immediately readable. Whenever I see then, I look for the assignment.